core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -784,3 +823,208 @@ impl RecoverableVertexStore for RocksDBStore {
         self.db.get_cf(self.cf_handle(&VertexStore), []).unwrap()
     }
 }
+
+pub struct WriteableRocksDBAccountChangeIndex<'a> {
+    store: &'a mut RocksDBStore,
+}
+
+impl<'a> WriteableRocksDBAccountChangeIndex<'a> {
+    fn batch_update_from_receipt(
+        &self,
+        batch: &mut WriteBatch,
+        state_version: u64,
+        receipt: &LedgerTransactionReceipt,
+    ) {
+        for (address, _) in receipt.state_update_summary.balance_changes.iter() {
+            if !address.is_account() {
+                continue;
+            }
+            let mut key = address.to_vec();
+            key.extend(state_version.to_be_bytes());
+            batch.put_cf(
+                self.store.cf_handle(&AccountChangeAtStateVersionSet),
+                key,
+                [],
+            );
+        }
+    }
+
+    fn batch_update_from_committed_transaction(
+        &self,
+        batch: &mut WriteBatch,
+        transaction_bundle: &CommittedTransactionBundle,
+    ) {
+        let state_version = transaction_bundle.2.state_version;
+
+        self.batch_update_from_receipt(batch, state_version, &transaction_bundle.1);
+
+        batch.put_cf(
+            self.store.cf_handle(&AccountChangeIndexExt),
+            "current_state_version".as_bytes(),
+            state_version.to_be_bytes(),
+        );
+    }
+
+    fn update_from_store(&mut self, start_state_version_inclusive: u64, limit: u64) -> u64 {
```

Comment:
It feels more natural for this code to live in a separate re-indexing utility, which is invoked at start-up.

Also this functionality would need better logging - because this process could take a long time.